sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -17,13 +17,36 @@ namespace Azure.AI.OpenAI
     /// <summary> Azure OpenAI APIs for completions and search. </summary>
     public partial class OpenAIClient
     {
+        private const string publicOpenAIVersion = "1";
+        private const string publicOpenAIEndpoint = $"https://api.openai.com/v{publicOpenAIVersion}";
+
+        /// <remarks> This key is used to connect to a non-Azure OpenAI resource created directly with OpenAI.
+        /// For Azure OpenAI resources as created and maintained in Azure Portal, use a constructor that provides
+        /// an Azure resource endpoint and credential, instead. </remarks>
+        private string PublicOpenAIToken { get; } = "";
+
+        /// <summary> Initializes a instance of OpenAIClient using the public OpenAI endpoint. </summary>
+        /// <param name="openAIAuthToken "> A credential used to authenticate to an Azure Service. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="openAIAuthToken"/> is null. </exception>
+        public OpenAIClient(string openAIAuthToken) : this(openAIAuthToken, new OpenAIClientOptions())
+        {
+        }
+
+        /// <summary> Initializes a instance of OpenAIClient using the public OpenAI endpoint. </summary>
+        /// <param name="openAIAuthToken "> String token to generate a token credential </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="openAIAuthToken"/> is null. </exception>
+        public OpenAIClient(string openAIAuthToken, OpenAIClientOptions options) : this(new Uri(publicOpenAIEndpoint), CreateDelegatedToken(openAIAuthToken), options)
+        {
+            PublicOpenAIToken = openAIAuthToken;
+        }
+
         /// <summary> Return the completion for a given prompt. </summary>
         /// <param name="deploymentId"> Deployment id (also known as model name) to use for operations </param>
         /// <param name="prompt"> Input string prompt to create a prompt completion from a deployment. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         public virtual async Task<Response<Completions>> GetCompletionsAsync(string deploymentId, string prompt, CancellationToken cancellationToken = default)
         {
-            Argument.AssertNotNullOrEmpty(deploymentId, nameof(deploymentId));
```

Comment:
I assume we are removing this because its not needed if you construct with an openai auth token?  This is odd to have a required param and force customers to pass null.  I wonder if we should pull in the change to move deploymentid to a client level parameter and require it on the non openai auth token ctors so that this is more straight forward?

Reply:
I think pulling that change into main would be reasonable, as it would certainly make usage here more straightforward.

Reply:
Agreed. It's not great that we have a "garbage" parameter. With a client-level default available, I think the overall decision flow would be something like:
`GetCompletions(string deploymentId, ...)` 
- Throws if the client was configured using non-Azure OpenAI information (API key)
- For clients properly configured to use Azure OpenAI, uses the method-provided deployment ID irrespective of what client-level default was provided
- Not null or empty stays enforced

`GetCompletions(string prompt / CompletionsOptions ...) // no deployment ID`
- Throws if the client was configured for Azure OpenAI *and* no client-level default deployment was provided
  - It'd be nice to someday have default deployment selection be an option, so maybe we could be flexible here
- Ignores the concept of deployment entirely for clients configured for non-Azure OpenAI (correctly)
- Uses the client-level default deployment ID for Azure OpenAI

I'm not sure we should wholesale *move* deployment ID specification to the client, though (vs. having a default setting) -- that could end up pretty clunky to use for scenarios where you want to use multiple models (or, if we don't provide affordances, multiple capabilities between completions/embeddings/chat/etc.)